### PR TITLE
Add ARCH_IS_WSL, to detect Windows Subsystem for Linux

### PR DESCRIPTION
**Pull request by ChrisJefferson**
This lets us detect the "Windows Subsystem for Linux".

I realise in some ways the naming has gotten weird now, we call Cygwin (one linux in windows) "ARCH_IS_WINDOWS", which WSL is not. While that is weird, I think it makes sense, WSL is a complete install of Linux (usually Ubuntu).

The only time we need ARCH_IS_WSL is when we want to open GUI programs -- I've added help viewers (unfortunately it doesn't seem possible to open either anchors in HTML, which makes this much less useful. I spent some time searching the internet, but everything people say seems to not work in practice, except making a new webpage which just forwards to the anchor, and I didn't want to do that as it would create lots of new temp files/directories).

I plan on using this in the 'profiling' package too, to automatically open HTML pages (I originally wrote this in that package, then thought I might be generally useful).

### CODE REVIEW

File: src/sysfiles.c [hunk 1]
```diff
@@ -339,6 +340,18 @@ static Obj FuncCrcString(Obj self, Obj str)
     return INTOBJ_INT(((Int4) crc) >> 4);
 }
 
+// Get OS Kernel version. Used to discover if GAP is running inside
+// 'Windows Subystem for Linux'
+Obj SyGetOsRelease(void)
+{
+    struct utsname buf;
+    if (!uname(&buf)) {
```
Comment (fingolfin) [inline src/sysfiles.c:348]: If you wrap uname, how about going the full way and return a record with all components? And then just return `fail` if there is an error?

File: lib/system.g [hunk 2]
```diff
@@ -564,6 +564,33 @@ BIND_GLOBAL("ARCH_IS_UNIX",function()
...
+#F  ARCH_IS_WSL()
+##
+##  <#GAPDoc Label="ARCH_IS_WSL">
+##  <ManSection>
+##  <Func Name="ARCH_IS_WSL" Arg=''/>
+##
+##  <Description>
+##  tests whether &GAP; is running on a Windows system inside the
+##  'Windows Subsystem for Linux'. Note that in this case
+##  <Ref Func="ARCH_IS_UNIX"/> will be <C>true</C>, and in most situations
```
Comment (wilfwilson) [inline lib/system.g:578]:
```suggestion
##  <Ref Func="ARCH_IS_UNIX"/> will be <K>true</K>, and in most situations
```

File: lib/helpview.gi [hunk 3]
```diff
@@ -198,48 +198,78 @@ elif ARCH_IS_MAC_OS_X() then
...
-    Exec(Concatenation("netscape -remote \"openURL(file:", url, ")\""));
-  end
-  );
+  # Graphical systems handled differently in WSL and standard Linux
+  if ARCH_IS_WSL() then
+    HELP_VIEWER_INFO.browser := rec(
+    type := "url",
+    show := function( url )
+      # Ignoring part of the URL after '#' since we are unable
+      # to navigate to the precise location on Windows
```
Comment (wilfwilson) [inline lib/helpview.gi:210]: That's weird, and unfortunate.

**Review by wilfwilson — APPROVED** [review]
Nice. I've got no suggestions on the code but I do have a cosmetic nitpick on the doc.

**Comment by fingolfin** [comment]
Just wondering, how is this related to PR #3939?

**Comment by ChrisJefferson** [comment]
This is unrelated to cygwin changes (although in the long term, the plan is the two distros will come closer together. This will be one place where they differ, as they have different ways of spawning files in windows)

File: src/sysfiles.c (hunk 1 above)
Comment (fingolfin) [reply src/sysfiles.c:348]: (to be clear, this is not a change required for this PR to be merged. You might want to apply the suggestion by Wilf, though)

**Review by fingolfin — APPROVED** [review]

File: lib/system.g (hunk 2 above)
Comment (ChrisJefferson) [reply lib/system.g:578]: I'm happy to fix it, although I coped what I wrote from about 20 lines higher up (I'll fix that too)

**Comment by ChrisJefferson** [comment]
Hopefully all cleaned up now. I decided not to fix other <C>true</C>, I thought they could be sweeped (also <C>false</C>) in another PR.

**Comment by wilfwilson** [comment]
Sure, thanks Chris. I'll merge this when the tests pass.

**Pull request merged**